File: Frontend/bleatf/app/lib/NotificationService.ts
// NotificationService for sending alerts and messages
export class NotificationService {
  static sendAlert(alert: any): boolean {
    console.log('Notification: Alert sent', alert);
    return true;
  }

  static sendMessage(message: any): boolean {
    console.log('Notification: Message sent', message);
    return true;
  }

  static notifyParent(parent: any, alert: any): void {
    console.log(`Notification sent to parent ${parent.name}:`, alert);
  }
}

File: Backend/src/main/java/com/bleat/services/NotificationService.java
package com.bleat.services;

import com.bleat.models.Alert;
import com.bleat.models.Message;
import com.bleat.models.Parent;

public class NotificationService {
    private static NotificationService instance = new NotificationService();

    private NotificationService() {}

    public static NotificationService getInstance() {
        return instance;
    }

    public boolean sendAlert(Alert alert) {
        // placeholder — persist to DB or push notification
        System.out.println("NotificationService: sending alert: " + alert);
        return true;
    }

    public boolean sendMessage(Message message) {
        System.out.println("NotificationService: sending message: " + message);
        return true;
    }

    public void notifyParent(Parent parent, Alert alert) {
        if (parent == null) return;
        parent.receiveAlert(alert);
    }
}
